core/src/tangram.cpp

```diff
@@ -98,11 +66,27 @@ void initialize(const char* _scenePath) {
     // Label setup
     m_labels = std::make_unique<Labels>();
 
+    // Create an AsyncWorker
+    m_asyncWorker = std::make_unique<AsyncWorker>();
+
     LOG("finish initialize");
 
 }
 
-void setScene(std::shared_ptr<Scene>& _scene) {
+Map::~Map() {
+    // Explicitly destroy all member objects so that we have a chance
+    // to run any resulting jobs sent to the JobQueue.
+    m_asyncWorker.reset();
+    m_inputHandler.reset();
+    m_labels.reset();
+    m_view.reset();
+    m_nextScene.reset();
+    m_tileManager.reset();
+    m_tileWorker.reset();
+    JobQueue::runJobsForCurrentThread();
```

Comment:
K, tested this, because of [line 218](https://github.com/tangrams/tangram-es/blob/826fa42f774343ee1a86eb1c460f713957943c3b/core/src/tangram.cpp#L218) and the `if block` at [line 225](https://github.com/tangrams/tangram-es/blob/826fa42f774343ee1a86eb1c460f713957943c3b/core/src/tangram.cpp#L225), handles this issue.

`m_nextScene` is copied as `scene` down the async Task and hence is still valid when this Task is run from the destructor. But since `m_nextScene` is reset, this task simply returns. And same applied to tasks created from `Map::applySceneUpdates`.